### ISSUE TITLE
Issues in cw_delay(...) function

### ISSUE THREAD

**Issue by abhisrkckl** [nanograv/enterprise_extensions#12]
I was trying to search for single source GWs using the circular template. I encountered the following issues in models.py .

1. Lines 632-636 should appear before line 622. Some of the variables here are used before they are defined.

2. Line 656 - The signature of the utils.create_gw_antenna_pattern(...) function used here is incorrect. 

3. Line 661 - The code fails when pdist is None.

4. Line 719 - The name of the boolean variable should be psrTerm rather than inc_psr_term. 

5. The cw_delay(...) function needlessly computes the pulsar term even when psrTerm is set to False.

**Comment by stevertaylor** [comment]
Hi! Sorry about the delay. The code was just updated to address some of these faults as well as others. Would you mind taking a look to see if the new changes help?

**Comment by bencebecsy** [comment]
I'm also having issues related to the pdist variable (just wanted to open an issue when I've realized this one). I've just updated my enterprise and enterprise_extension installation and the issue remains.

I set up my pta with cw_delay() and CWSignal() (plus noise models), and when I call the lnlikelihood I get the following error:
`TypeError: cw_delay() got multiple values for argument 'cos_gwtheta'`

I did some inspection on a previous version, and I think this happens because the pdist parameter is in confilct with something else (maybe pulsar object has an attribute called pdist?) and this results somehow in giving a value to cos_gwtheta (the first keyword argument) twice, because giving values to positional arguments flows over and starts to give values to keyword arguments.

In a previous version I managed to solve this by just renaming pdist to p_dist. I see something like this happened in the repo as well, but I see that there is still a pdist parmaeter, and the problem remains for me.
